gemrb/plugins/TLKImporter/TLKImporter.cpp

```diff
@@ -305,6 +305,16 @@ String TLKImporter::ResolveTags(const String& source)
 			} else {
 				dest.append(str);
 			}
+		} else if (srcch == L'%' && i + 1 < strLen - 1) {
+			// also resolve format strings
+			// we assume they are % followed by a single character
+			auto nextch = source[i + 1];
+			if (!isspace(nextch)) {
+				dest.append(L"{}");
+				++i;
+			} else {
+				dest.push_back(srcch);
+			}
```

Comment:
This is very fragile, but will work with the limited original set of strings with %d.

Reply:
yes, I didn't feel the need to be more robust/complicated. unless they used more than a simple type specifier this should work fine.

Reply:
We need to take care not to enable this path too often, which I'm afraid we do since cdd6940. That's the real danger, treating legitimate % use as tokens.

Reply:
I can't possibly see how cdd6940 leads to this being evaluated too frequently

I did think of legitimate '%' characters and thought it unlikely that legitimate ones would be immediately followed by a non whitespace/eof. Obviously if we can do a scan to verify that we should.

Reply:
True, it was just a reminder — it was bound to happen as soon as it was put in with token resolution. There are tons of uses with non-whitespace, notably punctuation, then missing spaces producing %c, %D and even %^ (for something we don't use). 
Matching %d is safe though and hardcoding it doesn't matter, since regular tokens cover the same functionality, if anyone ever needed it.

Reply:
bah, I just pushed a change for checking punctuation before reading this.

I'll change it to match %d only